src/components/Camera/AICamera/AICamera.js

```diff
@@ -171,14 +171,22 @@ const AICamera = ( {
   const handleTakePhoto = useCallback( async ( ) => {
     await logStage( sentinelFileName, "take_photo_start" );
     setHasTakenPhoto( true );
-    setAICameraSuggestion( result );
+    // this feels a little duplicative, but we're currently using aICameraSuggestion
+    // to show the loading screen in Suggestions *without* setting an observation.taxon,
+    // and we're using visionResult to populate ObsEdit *with* the taxon
+    // before aICameraSuggestion has finished being stored.
+    // would be nice to refactor and set this more uniformly once the UX is more stable
+    // and we're fully certain we don't want to populate observation.taxon on Suggestions -> ObsEdit
+    setAICameraSuggestion( showPrediction && result );
```

Comment:
I think we should not mix types here as long as useStore is returning the `any` type for all state. It makes it more predictable to use aiCameraSuggestion from state in knowing that it can only be of one type. So, eg if showPrediction is false we store a boolean and otherwise a result object (or null). So, here if I am not mistaken we could change to `showPrediction && setAICameraSuggestion( result );` instead.